File: 45. Jump Game II.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
        int step = 0;
        int endPos = 0;
        int maxPos =0;
        for(int i = 0; i < nums.size() - 1; i++) {
            maxPos = max(maxPos, nums[i] + i);
            if(i == endPos) {
                endPos = maxPos;
                step++;
            }
        }
        return step;
    }
};

File: 45. Jump Game II.java
class Solution {
    public int jump(int[] nums) {
        int n = nums.length;
        int[] f = new int[n];
        for(int i = 1, j = 0; i < n; i++) {
            while(j + nums[j] < i) {
                j++;
            }
            f[i] = f[j] + 1;
        }
        return f[n - 1];
    }
}